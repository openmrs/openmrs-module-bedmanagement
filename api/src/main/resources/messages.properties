${project.parent.artifactId}.title=bedmanagement Module
${project.parent.artifactId}.admissionLocations=Admission Locations
${project.parent.artifactId}.bedTypes=Bed Types
${project.parent.artifactId}.bedTags=Bed Tags