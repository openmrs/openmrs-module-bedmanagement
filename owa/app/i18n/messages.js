export default {
    en: {
        ADMISSION_LOCATIONS: 'Admission Locations',
        BED_TYPES: 'Bed Types',
        BED_TAGS: 'Bed Tags',
        LOGOUT: 'Logout',
        CHOOSE_LANGUAGE: 'Choose language',
        ROW_REQUIRED_MSG: 'Row is required field',
        COLUMN_REQUIRED_MSG: 'Column is required field',
        ROW_SHOULD_GREATER_THAN_ZERO: 'Row value should be greater than 0',
        ROW_SHOULD_GREATER_THAN_LAYOUT_ROW: 'Row value should not be greater than layout row size',
        COLUMN_SHOULD_GREATER_THAN_ZERO: 'Column value should be greater than 0 and less than 11',
        COLUMN_SHOULD_GREATER_THAN_LAYOUT_COLUMN: 'Row value should not be greater than layout row size',
        FIX_ERROR_MSG: 'Fix error before submit',
        ADMISSION_LOCATION_SAVE_MSG: 'Admission location save successfully',
        BED_LAYOUT_SAVE_MSG: 'Admission location bed layout save successfully',
        BED_TAG_SAVE_MSG: 'Bed Tag save successfully',
        BED_TYPE_SAVE_MSG: 'Bed Type save successfully',
        BED_SAVE_MSG: 'Bed save successfully',
        DELETE_ADMISSION_LOCATION_CONFIRM_MESSAGE:
            'Are you sure you want to delete admission location {location_name}?',
        DELETE_BED_CONFIRM_MESSAGE: 'Are you sure you want to delete bed number {bed_number}?',
        DELETE_BED_TAG_CONFIRM_MSG: 'Are you sure you want to delete bed Tag {bed_tag_name}?',
        DELETE_BED_TYPE_CONFIRM_MSG: 'Are you sure you want to delete bed type {bed_type_name}?',
        DELETE_SUCCESSFULLY: 'Delete successfully',
        SUCCESS: 'Success',
        ERROR: 'Error',
        INFO: 'Info',
        SAVE: 'Save',
        SAVING: 'Saving...',
        CANCEL: 'Cancel',
        ADD: 'Add',
        EDIT: 'Edit',
        DELETE: 'Delete',
        ADD_BED: 'Add Bed',
        ROWS: 'Rows',
        ROW: 'Row',
        COLUMNS: 'Columns',
        COLUMN: 'Column',
        SET_LAYOUT: 'Set Layout',
        LOCATION: 'Location',
        BED: 'Bed',
        BED_NUMBER: 'Bed Number',
        BED_TYPE: 'Bed Type',
        WARD: 'Ward',
        PARENT_LOCATION: 'Parent Location',
        NONE: 'None',
        NAME: 'Name',
        DESCRIPTION: 'Description',
        DISPLAY_NAME: 'Display Name',
        ACTION: 'Action',
        ADD_NEW: 'Add New',
        BED_TAG: 'Bed Tag',
        EXISTING_BED_TYPES: 'Existing Bed Types',
        EXISTING_BED_TAGS: 'Existing Bed Tags'
    }
};
