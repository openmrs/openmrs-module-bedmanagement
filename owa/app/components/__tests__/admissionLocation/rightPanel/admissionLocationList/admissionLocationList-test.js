import React from 'react';
import {shallow, mount} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';
import {IntlProvider} from 'react-intl';

import AdmissionLocationList from 'components/admissionLocation/rightPanel/admissionLocationList';
import admissionLocationFunctionsMock from 'components/__mocks__/admissionLocationFunctions-mock';
import messages from 'i18n/messages';

const intlProvider = new IntlProvider({locale: 'en', messages: messages['en']}, {});
const {intl} = intlProvider.getChildContext();
const testData = {
    props: {
        admissionLocationFunctions: admissionLocationFunctionsMock,
        admissionLocations: admissionLocationFunctionsMock.getAdmissionLocations()
    },
    context: {
        intl: intl
    }
};

describe('AdmissionLocationList', () => {
    it('Should render location list properly', () => {
        const generalWardLocationList = shallow(
            <AdmissionLocationList
                activeUuid="baf7bd38-d225-11e4-9c67-080027b662ec"
                admissionLocationFunctions={testData.props.admissionLocationFunctions}
            />,
            {context: testData.context}
        );

        expect(generalWardLocationList.find('LocationBlock').length).toBe(2);
        expect(shallowToJson(generalWardLocationList)).toMatchSnapshot();

        const labourWardLocationList = shallow(
            <AdmissionLocationList
                activeUuid="bb0e512e-d225-11e4-9c67-080027b662ec"
                admissionLocationFunctions={testData.props.admissionLocationFunctions}
            />,
            {context: testData.context}
        );

        expect(labourWardLocationList.find('LocationBlock').length).toBe(0);
        expect(shallowToJson(labourWardLocationList)).toMatchSnapshot();
    });

    it('Should trigger event handler', () => {
        const spyOnAddWardClickHandler = jest.spyOn(AdmissionLocationList.prototype, 'addWardClickHandler');
        const spyOnSetState = jest.spyOn(testData.props.admissionLocationFunctions, 'setState');
        const generalWardLocationList = mount(
            <AdmissionLocationList
                activeUuid="baf7bd38-d225-11e4-9c67-080027b662ec"
                admissionLocationFunctions={testData.props.admissionLocationFunctions}
            />,
            {context: testData.context}
        );

        generalWardLocationList.find('.fa-plus').simulate('click');
        expect(spyOnAddWardClickHandler).toHaveBeenCalled();
        expect(spyOnSetState).toHaveBeenCalledWith({
            activePage: 'addEditLocation',
            pageData: {
                operation: 'add'
            },
            activeUuid: 'baf7bd38-d225-11e4-9c67-080027b662ec'
        });
    });
});
