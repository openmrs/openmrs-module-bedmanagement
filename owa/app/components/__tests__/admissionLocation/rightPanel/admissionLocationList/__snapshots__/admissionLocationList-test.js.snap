// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdmissionLocationList Should render location list properly 1`] = `
<div
  className="main-container"
>
  <Breadcrumb
    activeUuid="baf7bd38-d225-11e4-9c67-080027b662ec"
    admissionLocationFunctions={
      Object {
        "getActiveLocationUuid": [Function],
        "getAdmissionLocationByUuid": [Function],
        "getAdmissionLocations": [Function],
        "getBedTypes": [Function],
        "getChildAdmissionLocations": [Function],
        "getParentAdmissionLocation": [Function],
        "getState": [Function],
        "getVisitLocations": [Function],
        "notify": [Function],
        "reFetchAllAdmissionLocations": [Function],
        "setActiveLocationUuid": [Function],
        "setState": [Function],
      }
    }
  />
  <div
    className="main-block"
  >
    <div>
      <LocationBlock
        admissionLocation={
          Object {
            "description": "General Ward room number 2",
            "isHigherLevel": false,
            "isOpen": false,
            "name": "General Ward - Room 2",
            "parentAdmissionLocationUuid": "baf7bd38-d225-11e4-9c67-080027b662ec",
            "uuid": "e48fb2b3-d490-11e5-b193-0800270d80ce",
          }
        }
        admissionLocationFunctions={
          Object {
            "getActiveLocationUuid": [Function],
            "getAdmissionLocationByUuid": [Function],
            "getAdmissionLocations": [Function],
            "getBedTypes": [Function],
            "getChildAdmissionLocations": [Function],
            "getParentAdmissionLocation": [Function],
            "getState": [Function],
            "getVisitLocations": [Function],
            "notify": [Function],
            "reFetchAllAdmissionLocations": [Function],
            "setActiveLocationUuid": [Function],
            "setState": [Function],
          }
        }
        key="e48fb2b3-d490-11e5-b193-0800270d80ce"
      />
      <LocationBlock
        admissionLocation={
          Object {
            "description": "General Ward of second floor",
            "isHigherLevel": false,
            "isOpen": false,
            "name": "General Ward - floor-2",
            "parentAdmissionLocationUuid": "baf7bd38-d225-11e4-9c67-080027b662ec",
            "uuid": "baf83667-d225-11e4-9c67-080027b662ec",
          }
        }
        admissionLocationFunctions={
          Object {
            "getActiveLocationUuid": [Function],
            "getAdmissionLocationByUuid": [Function],
            "getAdmissionLocations": [Function],
            "getBedTypes": [Function],
            "getChildAdmissionLocations": [Function],
            "getParentAdmissionLocation": [Function],
            "getState": [Function],
            "getVisitLocations": [Function],
            "notify": [Function],
            "reFetchAllAdmissionLocations": [Function],
            "setActiveLocationUuid": [Function],
            "setState": [Function],
          }
        }
        key="baf83667-d225-11e4-9c67-080027b662ec"
      />
      <span
        className="btn btn-primary"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="icon fa fa-plus"
        />
         Add Ward
      </span>
    </div>
  </div>
</div>
`;

exports[`AdmissionLocationList Should render location list properly 2`] = `
<div
  className="main-container"
>
  <Breadcrumb
    activeUuid="bb0e512e-d225-11e4-9c67-080027b662ec"
    admissionLocationFunctions={
      Object {
        "getActiveLocationUuid": [Function],
        "getAdmissionLocationByUuid": [Function],
        "getAdmissionLocations": [Function],
        "getBedTypes": [Function],
        "getChildAdmissionLocations": [Function],
        "getParentAdmissionLocation": [Function],
        "getState": [Function],
        "getVisitLocations": [Function],
        "notify": [Function],
        "reFetchAllAdmissionLocations": [Function],
        "setActiveLocationUuid": [Function],
        "setState": [Function],
      }
    }
  />
  <div
    className="main-block"
  >
    <BedLayout
      activeUuid="bb0e512e-d225-11e4-9c67-080027b662ec"
      admissionLocationFunctions={
        Object {
          "getActiveLocationUuid": [Function],
          "getAdmissionLocationByUuid": [Function],
          "getAdmissionLocations": [Function],
          "getBedTypes": [Function],
          "getChildAdmissionLocations": [Function],
          "getParentAdmissionLocation": [Function],
          "getState": [Function],
          "getVisitLocations": [Function],
          "notify": [Function],
          "reFetchAllAdmissionLocations": [Function],
          "setActiveLocationUuid": [Function],
          "setState": [Function],
        }
      }
    />
  </div>
</div>
`;
